step0_processor.py

```diff
@@ -0,0 +1,513 @@
+import ee
+from datetime import datetime, timedelta
+
+ee.Initialize()
+
+
+def generate_asset_mosaic_for_single_date(day_to_process: datetime, export_to_drive: bool = False,
```

Comment:
Question: 
- This function is made for S2-L2A, and not for S2-L1C, so where would the switch be? ow r will it be a different function?
- Where are the switches as defined inhttps://code.earthengine.google.com/a08635b7f3c83b082c35023a72743ec4 line 46 to line 69. Does it makes sense to have them as well in config file in the product definition?